app/src/main/java/org/mozilla/tv/firefox/ScreenController.kt

```diff
@@ -169,19 +170,21 @@ class ScreenController {
         if (_currentActiveScreen.value == ActiveScreen.WEB_RENDER) {
             if (webRenderFragment.onBackPressed()) return true
         }
-        val transition = ScreenControllerStateMachine.getNewStateBackPress(_currentActiveScreen.value!!, isOnHomeUrl(fragmentManager))
+        val transition = ScreenControllerStateMachine.getNewStateBackPress(_currentActiveScreen.value!!, canGoBack())
         return handleTransitionAndUpdateActiveScreen(fragmentManager, transition)
     }
 
     fun handleMenu(fragmentManager: FragmentManager): Boolean {
-        val transition = ScreenControllerStateMachine.getNewStateMenuPress(_currentActiveScreen.value!!, isOnHomeUrl(fragmentManager))
+        val transition = ScreenControllerStateMachine.getNewStateMenuPress(_currentActiveScreen.value!!, isOnHomeUrl())
         return handleTransitionAndUpdateActiveScreen(fragmentManager, transition)
     }
 
-    private fun isOnHomeUrl(fragmentManager: FragmentManager): Boolean {
-        // TODO: Would be more correct to get this from the model rather than the Fragment.
-        val webRenderFragment = fragmentManager.webRenderFragment()
-        return webRenderFragment.session.url == URLs.APP_URL_HOME
+    private fun canGoBack(): Boolean {
+        return sessionRepo.state.value?.backEnabled == true
```

Comment:
It's good to gate this on back stack - one thing this does do is add another layer on top of the `canGoBackTwice` which still bothers me, but that's a bigger fix (and depends on https://github.com/mozilla-mobile/android-components/issues/1993)

Reply:
I think it should be roughly independent of `canGoBackTwice`, right?  Regardless of how we calculate when the back button is enabled, once it's not we should exit the app.  To me that seems like a relatively generic seam.